### ISSUE TITLE
stuck trying to make both a rmd 2 pdf and a Rd2pdf work

### ISSUE THREAD

**Issue by kforner** [rocker-org/rocker-versioned2#352]
Hi,
I have a docker image based on a recent `rocker/verse:4.1.2`

I have a TeX template, used to implement a custom rmarkdown document format, that renders to pdf.
If I render a rmd with that format, I get a TeX error like:
```
[352] "! LaTeX Error: File `courier.sty' not found."  
```
I also have a bunch of R packages, whose manuals build happily using `R CMD Rd2pdf`.

Now, If I install the `psnfss` texlive package using `sudo tlmgr update --self && sudo tlmgr install psnfss`, the rmd renders successfully.
But the manuals now fail with messages like:
```
! Font T1/ptm/b/n/20.74=ptmb8t at 20.74pt not loadable: Metric (TFM) file not f
ound.
<to be read again> 
                   relax 
l.10 {\textbf{\huge Package `xxxx'}
                                    }
Missing character: There is no P in font nullfont!
Missing character: There is no a in font nullfont!
Missing character: There is no c in font nullfont!
Missing character: There is no k in font nullfont!
Missing character: There is no a in font nullfont!
Missing character: There is no g in font nullfont!
Missing character: There is no e in font nullfont!
Missing character: There is no ` in font nullfont!
Missing character: There is no q in font nullfont!
Missing character: There is no b in font nullfont!
Missing character: There is no r in font nullfont!
Missing character: There is no m in font nullfont!
Missing character: There is no d in font nullfont!
Missing character: There is no ' in font nullfont!
! Font T1/ptm/m/n/12=ptmr8t at 12.0pt not loadable: Metric (TFM) file not found

```

I tried many many things but I can not manage to make both work at the same time. 
It worked previously using an (older) tinytex-based setup.

Any hints ? 
Thanks.

**Comment by eddelbuettel** [comment]
Maybe see the discussion (just updated yesterday) in [this sibbling issue](https://github.com/rocker-org/rocker-versioned2/issues/349) and especially Carl's comment [here](https://github.com/rocker-org/rocker-versioned2/issues/349#issuecomment-1039532278) ?

**Comment by kforner** [comment]
the latest rocker/verse:4.1.2 does not use the versioned CTAN:
```
%docker run --rm -ti rocker/verse:4.1.2 env | grep CTAN_REPO
CTAN_REPO=https://mirror.ctan.org/systems/texlive/tlnet
```
Do you think that my problem could be due to an outdated texlive install as compared to the packages pulled out by `tlmgr` ?


From what I read in some internet posts, the general recommendation is to install APT packages such as **texlive-fonts-extra** or **texlive-fonts-recommended**. 
Because the rocker/verse texlive is not APT-based, this is not really an option. 
I have no clue about what is the texlive counterpart of installing `texlive-fonts-extra`. Do you think it could be related to my problem ?

**Comment by eddelbuettel** [comment]
I really don't know. I am torn on the issue.  I really *love* the fact that my normal (desktop and alike) machine(s) all LaTeX away til the cows come home: easy, reliably, and without fuzz. Also extensible with local font packages as .deb files I added etc.

At the same time I see the large footprint.  So in 'cloudy' deployment i.e. when I do CI and alike I never install any latex and generally skip vignettes and manuals.  That is just my preference -- I take "faster" and "not failing for random reasons" here.   So with that I generally do not have the inclination to learn to live with a third option that may be seen as being more fragile and changing.  Again, that is just me.  But I pointed to Carl's comment to show you that I am not entirely alone here.

And because "yes we can" I also at some point created a Rocker variant with all of it on it as I do own a few packages that ship markdown templates and styles and needed the testing.  Maybe you need to go down a similar route.  

Long story short you if you want LaTeX / texlive setups very specific to your use case you may have to put the ellbow grease to debug its creato

**Comment by kforner** [comment]
> And because "yes we can" I also at some point created a Rocker variant with all of it on it as I do own a few packages that ship markdown templates and styles and needed the testing. Maybe you need to go down a similar route.

That's what I did previously (based on tinytex), and that's why I was so happy when I discovered that `rocker/verse` was providing a TeX installation ! ;)

I've seen that you said somewhere, and I agree, that the ideal solution would be a debian-based installation, but less monolithic. 
I suppose it did not get traction ?

**Comment by kforner** [comment]
This seemed to fix my problem:
`sudo tlmgr install collection-fontsrecommended`
I suppose this is the counter-part for texlive-fonts-recommended I was looking for.

**Comment by kforner** [comment]
Actually 
`sudo tlmgr install times` was enough to fix Rd2pdf.

**Comment by cboettig** [comment]
Thanks @kforner .  

> the latest rocker/verse:4.1.2 does not use the versioned CTAN:

Yeah, following our conventions, the current release (aka `latest` doesn't use pinned versions of anything).  (Also the maintainer of the CTAN version snapshots has suggested to us that pinned version server isn't built to handle the same volume of requests as the standard release mirrors, but `latest` gets hit most anyway).

yeah, I've hit issues with tinytex not being able to tlmgr install some font files automatically.  Thanks for sharing your solution!

(sometimes I just install texlive, and some days I think with modern css that using latex is so archaic that I can live without it but that's so not true).

**Comment by kforner** [comment]
thanks both for your help!
